hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/HFileOutputFormat2.java

```diff
@@ -254,16 +261,21 @@ public void write(ImmutableBytesWritable row, V cell) throws IOException {
         byte[] tableNameBytes = null;
         if (writeMultipleTables) {
           tableNameBytes = MultiTableHFileOutputFormat.getTableName(row.get());
-          tableNameBytes = TableName.valueOf(tableNameBytes).toBytes();
+          tableNameBytes = (writeToTableWithNamespace) ?
+            TableName.valueOf(tableNameBytes).getNameWithNamespaceInclAsString()
+              .getBytes(Charset.defaultCharset()) :
+            TableName.valueOf(tableNameBytes).toBytes();
           if (!allTableNames.contains(Bytes.toString(tableNameBytes))) {
             throw new IllegalArgumentException(
               "TableName " + Bytes.toString(tableNameBytes) + " not expected");
           }
         } else {
           tableNameBytes = Bytes.toBytes(writeTableNames);
         }
+        String tableName = Bytes.toString(tableNameBytes);
```

Comment:
was there a reason to move this up here?

Reply:
There was a reason on why this was done. Can't recollect now. Logically it seems no difference to me.

Reply:
Agreed. It looks the same. I just like to minimize unnecessary diffs

Reply:
Let me move it back and run tests.